src/compiler/msvc.rs

```diff
@@ -173,53 +173,63 @@ pub fn parse_arguments(arguments: &[String]) -> CompilerArguments<ParsedArgument
     let mut show_includes = false;
 
     let mut it = arguments.iter().map(|i| {
-        if i.starts_with("/") {
-            format!("-{}", &i[1..])
+        if let Some(arg) = i.split_prefix("/") {
+            let mut dash = OsString::from("-");
+            dash.push(&arg);
+            dash
         } else {
-            i.to_string()
+            i.clone()
         }
     });
     while let Some(arg) = it.next() {
-        match &arg[..] {
-            "-c" => compilation = true,
-            v if v.starts_with("-Fo") => {
-                output_arg = Some(String::from(&v[3..]));
-            }
-            // Arguments that take a value.
-            "-FI" => {
+        if let Some(arg) = arg.split_prefix("-Fo") {
```

Comment:
I wonder if this code would be nicer if we refactored it similarly to the Rust code that splits it out into `(arg, val)` tuples? We'd have to do some additional fiddling to handle the goofy MSVC argument styles.